CMakeLists.txt

```diff
@@ -210,6 +210,18 @@ add_library(l3 OBJECT incorrect_source.cpp)
 endfunction()
 forbid_undefined_symbols()
 
+if (MINGW)
+  function(export_all_symbols)
+    unset(TMP)
+    add_linker_flag_if_supported(-Wl,--export-all-symbols TMP)
+    string(APPEND CMAKE_SHARED_LINKER_FLAGS ${TMP})
+    string(APPEND CMAKE_MODULE_LINKER_FLAGS ${TMP})
+    set(CMAKE_SHARED_LINKER_FLAGS ${CMAKE_SHARED_LINKER_FLAGS} PARENT_SCOPE)
+    set(CMAKE_MODULE_LINKER_FLAGS ${CMAKE_MODULE_LINKER_FLAGS} PARENT_SCOPE)
+  endfunction()
+  export_all_symbols()
```

Comment:
Why do we need a function?

Any benefits comparing to

```cmake
add_linker_flag_if_supported(-Wl,--export-all-symbols EXPORT_ALL_SYMBOLS_FLAG)
string(APPEND CMAKE_SHARED_LINKER_FLAGS ${EXPORT_ALL_SYMBOLS_FLAG})
string(APPEND CMAKE_MODULE_LINKER_FLAGS ${EXPORT_ALL_SYMBOLS_FLAG})
```

or even (yep, will execute the test twice, though, do we really care)?
```cmake
add_linker_flag_if_supported(-Wl,--export-all-symbols CMAKE_SHARED_LINKER_FLAGS)
add_linker_flag_if_supported(-Wl,--export-all-symbols CMAKE_MODULE_LINKER_FLAGS)
```

Reply:
@perfect-daemon did it this way due to

```
09:23 <wfaressuissia[m]> I want to keep that code wrapped into function since it will keep related calls together and will not pollute parent scope with temporary variables
09:25 <wfaressuissia[m]> and I prefer to wrap related code into function with verbose name export_all_symbols(...) instead of unwrapped code and single line comment
10:38 <wfaressuissia[m]> also with separate function there is no need to check parent scope for usage of the same temporary variables
```